packages/containers/src/List/List.sagas.js

```diff
@@ -0,0 +1,80 @@
+import { put, takeEvery, call } from 'redux-saga/effects';
+import { List } from 'immutable';
+import Connected from './List.connect';
+import Constants from './List.constant';
+import { getCollectionItems, sortCurried, filterList } from './utils';
+
+const getFilteredItems = filterListFn => (componentState, config, results) => {
+	const filterItemsCurried = sortCurried(componentState, config);
+	return filterItemsCurried(filterListFn(componentState, config, results));
+};
+
+export function* updateList({ props, context }) {
+	const collectionId = props.collectionId || 'default';
+	const state = context.store.getState();
+	const stateItems = getCollectionItems(state, collectionId);
+	const componentState = state.cmf.components.getIn(['Container(List)', collectionId]);
+
+	const items = stateItems || props.config.items || new List();
+	const results = getFilteredItems(filterList)(componentState, props.config, items);
```

Comment:
the currying was for the functions for the sort and filter directly not new top one :p
like this :

```javascript
// Here I assume 'stateItems' and 'props.config.items' are Immutable List, if not always the case, will need to check that before, an make sure we have at this point a strong consistent 'items' Immutable List define (empty or not but defined).
const items = stateItems || props.config.items || new List();
const filteredItem = searchQuery
    ? items.filter(containsQuery(searchQuery, columnConfig))
    : items;
// Here we do the same as for the filter, the 'sortCurriedToRename' need to return a function with the normal interface : (item1, item2) => compute boolean with access to sortBy, etc.
const sortedItems = filteredItem.sort(sortCurriedToRename(sortBy, sortAsc, columnsConfig));
```

I inverse the filter before the sort, I don't know if it's possible for the current case, but it make less computation to do this in this order. It should be possible unless if there is side effects somewhere.